mindsdb/integrations/handlers/merlion_handler/merlion_handler.py

```diff
@@ -0,0 +1,193 @@
+from enum import Enum
+import json
+import numpy as np
+import pandas as pd
+
+from mindsdb.integrations.libs.base import BaseMLEngine
+from mindsdb.utilities.log import log
+from .adapters import BaseMerlionForecastAdapter, DefaultForecasterAdapter, MerlionArguments, DefaultDetectorAdapter, \
+    SarimaForecasterAdapter, ProphetForecasterAdapter, MSESForecasterAdapter, IsolationForestDetectorAdapter, \
+    WindStatsDetectorAdapter, ProphetDetectorAdapter
+
+
+class DetectorModelType(Enum):
+    default = DefaultDetectorAdapter
+    isolation = IsolationForestDetectorAdapter
+    windstats = WindStatsDetectorAdapter
+    prophet = ProphetDetectorAdapter
+
+
+class ForecastModelType(Enum):
+    default = DefaultForecasterAdapter
+    sarima = SarimaForecasterAdapter
+    prophet = ProphetForecasterAdapter
+    mses = MSESForecasterAdapter
+
+
+class TaskType(Enum):
+    detector = DetectorModelType
+    forecast = ForecastModelType
+
+
+def is_invalid_type(name: str, type_class: Enum) -> bool:
+    if name is None:
+        return True
+    return not name in type_class._member_names_
+
+
+def enum_to_str(type_class: Enum) -> str:
+    all = []
+    for element in type_class:
+        all.append(element.name)
+    return "|".join(all)
+
+
+def to_ts_dataframe(df: pd.DataFrame, time_col=None) -> str:
+    columns = list(df.columns.values)
+    if time_col is not None :
+        if time_col not in columns:
+            raise Exception("invalid column name: " + time_col)
+        if df[time_col].dtype != np.datetime64:
+            try:
+                df.index = pd.to_datetime(df[time_col])
+            except Exception as e:
+                raise Exception("can not convert column to datetime: " + time_col + " " + str(e))
+    else:
+        datetime_cols = list(df.select_dtypes(include=["datetime"]).columns.values)
+        if len(datetime_cols) > 0:
+            time_col = datetime_cols[0]
+            df.index = pd.to_datetime(df[time_col])
+        if time_col is None:
+            raise Exception("can not find datetime column for time series")
+    df.drop(columns=[time_col], inplace=True)
```

Comment:
Side-effects like this are best avoided. Can you modify this method to explicitly return a new dataframe on which you perform all changes (i.e. modifying index, `pop()`ing columns, etc.)?

Reply:
ok, I will return a fully new dataframe. You are right, it's not wise to modify on the input.